dist/screeps.d.ts

```diff
@@ -1017,8 +1017,7 @@ interface SignDefinition {
 }
 interface StoreDefinition {
     [resource: string]: number;
-    energy?: number;
-    power?: number;
+    energy: number;
```

Comment:
When asserting for all `x` that `f[T](x) = <T>(x')`, we claim `f` to be complete, forgoing the notion that the code must verify the presence of the value before it may be utilized.

Observe the type of the value returned by selection of this missing property:
![image](https://cloud.githubusercontent.com/assets/434942/22631902/4b957c00-ebc9-11e6-8f3d-62d05186130b.png)

When we show that it may be undefined, we preserve the dictionary's nature as a partial function, and assert that the user of the value must ensure its presence beforehand:
![image](https://cloud.githubusercontent.com/assets/434942/22632003/4cc94bfa-ebca-11e6-99d4-21270a5c6a2f.png)

Note that this is then seen by the language as the correct type when guard clauses constrain potential interface surfaces:
![image](https://cloud.githubusercontent.com/assets/434942/22632034/a658134a-ebca-11e6-87cc-5284ee2cdd0a.png)

Finally, if we wish to assert that a property (e.g. `energy`) is always present, we can clarify the substate for that particular mapping:
![image](https://cloud.githubusercontent.com/assets/434942/22632336/366508ba-ebcd-11e6-9f53-e88b5fdfd24f.png)

Preserving semantics in types ensures that the code is consistent and logical, and any situation where you find yourself having to cast something without consideration of its existing type indicates a misrepresentation within the type system.